server/player/player.go

```diff
@@ -3109,3 +3224,11 @@ func (p *Player) resendBlock(pos cube.Pos, w *world.World) {
 func format(a []any) string {
 	return strings.TrimSuffix(strings.TrimSuffix(fmt.Sprintln(a...), "\n"), "\n")
 }
+
+// DimensionProvider provides access to all 3 dimensions.
+type DimensionProvider interface {
+	World() *world.World
+	Nether() *world.World
+	End() *world.World
+	Players() []*Player
+}
```

Comment:
This interface feels like it's very limiting for worlds, what happens when a server has multiworld support?

Reply:
I agree, but how do I determine which world to teleport the player to after death?

Reply:
I think the default world is appropriate for that, but maybe add a respawn event that lets you modify the new world and position?

Reply:
df already has a respawn handler
https://github.com/df-mc/dragonfly/blob/0637012e5ea28136785feb3f437de2fb7c4d2460/server/player/handler.go#L55

Reply:
Is there something I left to do?